src/components/NewDatePicker/index.native.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import {View, Keyboard} from 'react-native';
+import moment from 'moment';
+import _ from 'underscore';
+import TextInput from '../TextInput';
+import CONST from '../../CONST';
+import styles from '../../styles/styles';
+import {propTypes, defaultProps} from './datePickerPropTypes';
+import withKeyboardState, {keyboardStatePropTypes} from '../withKeyboardState';
+import CalendarPicker from '../CalendarPicker';
+
+const datepickerPropTypes = {
+    ...propTypes,
+    ...keyboardStatePropTypes,
+};
+
+const datePickerDefaultProps = {
+    ...defaultProps,
+};
+
+class NewDatePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isPickerVisible: false,
+            selectedDate: moment(props.value || props.defaultValue).toDate(),
```

Comment:
Why on native the `selectedDate` is initiated from props while on other platforms it's not.